Makefile

```diff
@@ -2,7 +2,7 @@ GOMOD ?= on
 GO ?= GO111MODULE=$(GOMOD) go
 
 #Don't enable mod=vendor when GOMOD is off or else go build/install will fail
```

Comment:
Currently it still applies, because if GO111MODULE is not set then using `-mod=anything` would fail. I guess we don't need to allow toggling GOMOD any more so I'll fix it.

Reply:
As information from: https://github.com/golang/go/wiki/Modules#faqs--possible-problems

- Double-check that modules are enabled by running go env to confirm it does not show an empty value for the read-only GOMOD variable.

- Note: you never set GOMOD as a variable because it is effectively read-only debug output that go env outputs.

Reply:
@jfkw ah, that's good background information. If GOMOD is read-only, then this variable was probably always useless in this make script since if it was unset in the go environment it wouldn't be able to override the setting here?

I'll update the commit message to reference that FAQ.